File: 66-PlusOne/plusOne.java
public class Solution {
    public int[] plusOne(int[] digits) {
        int j = digits.length -1;
        while(j>=0 && digits[j] == 9) {
            digits[j] = 0;
            j--;
        }
        
        if (j == -1) {
            int[] rslt = new int[digits.length+1];
            rslt[0] =1;
            return rslt;
        } else {
            digits[j]++;
            return digits;
        }
    }
}

File: 66-PlusOne/plusOne.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int carry=1;
        int x=1;
        vector<int> res;
        for(int i=digits.size()-1;i>=0;i--){
            int sum=digits[i]+carry;
            if(sum<10)  {carry=0;res.push_back(sum);}
            if(sum>=10)  {carry=1;res.push_back(sum%10);}
        }
        if(carry)   res.push_back(1);
        reverse(res.begin(),res.end());
        return res;
    }
};